packages/jsii-pacmak/lib/targets/dotnet.ts

```diff
@@ -126,44 +127,845 @@ export default class Dotnet extends Target {
+        } else {
+            this.emitProperty(cls, prop);
+        }
+    }
+
+    protected onUnionProperty(cls: spec.ClassType, prop: spec.Property, _union: spec.UnionTypeReference) {
+        this.emitProperty(cls, prop);
+    }
+
+    protected onBeginEnum(enm: spec.EnumType) {
+        const enumName = nameutils.convertTypeName(enm.name);
+        this.openFileIfNeeded(enumName, this.assembly.targets!.dotnet!.namespace, this.isNested(enm));
+        this.emitDocs(enm);
+        const jsiiAttribute = `[JsiiEnum(nativeType: typeof(${enumName}), fullyQualifiedName: \"${enm.fqn}\")]`;
+        // Emit Jsii Attribute
+        this.code.line(jsiiAttribute);
+        this.code.openBlock(`public enum ${enm.name}`);
+    }
+
+    protected onEndEnum(enm: spec.EnumType) {
+        this.code.closeBlock();
+        const enumName = nameutils.convertTypeName(enm.name);
+        this.closeFileIfNeeded(enumName, this.isNested(enm));
+    }
+
+    protected onEnumMember(_: spec.EnumType, member: spec.EnumMember) {
+        const enumMemberName = nameutils.convertEnumMemberName(member.name);
+        const jsiiAttribute = `[JsiiEnumMember(name: "${enumMemberName}")]`;
+        // Emit Jsii Attribute
+        this.code.line(jsiiAttribute);
+        // If we are on the last enum member, we don't need a comma
+        if (_.members.indexOf(member) !== (_.members.length - 1)) {
+            this.code.line(`${enumMemberName},`);
+        } else {
+            this.code.line(`${enumMemberName}`);
+        }
+    }
+
+    private emitMethod(cls: spec.Type, method: spec.Method, emitForProxyOrDatatype: boolean = false): void {
+        this.emitNewLineIfNecessary();
+        const returnType = method.returns ? this.toDotNetType(method.returns.type) : 'void';
+        const staticKeyWord = method.static ? 'static ' : '';
+
+        let overrideKeyWord = '';
+        let virtualKeyWord = '';
+
+        const definedOnAncestor = this.isMemberDefinedOnAncestor(cls as spec.ClassType, method);
+        // The method is an override if it's defined on the ancestor, or if the parent is a class and we are generating a proxy or datatype class
+        const overrides = (definedOnAncestor || (cls.kind === spec.TypeKind.Class && emitForProxyOrDatatype));
+        if (overrides) {
+            // Add the override key word if the method is emitted for a proxy or data type or is defined on an ancestor
+            overrideKeyWord = 'override ';
+        } else {
+            // Add the virtual key word if the method is abstract or not defined on an ancestor and we are NOT generating a proxy or datatype class
+            // Methods should always be virtual when possible
+            if ((method.abstract || !definedOnAncestor) && !emitForProxyOrDatatype) {
+                virtualKeyWord = 'virtual ';
+            }
+
+        }
+        const access = this.renderAccessLevel(method);
+        // TODO: handle async methods when impl calc-base and calc
```

Comment:
`async` can be a JavaScript-only keyword. No need to translate into C#'s version of `async/await`, if that's what you were planning. But, that also depends on support from the runtime, and I'm not 100% sure on what that does today. Ask Aaron.

Reply:
The generator/runtime does not support async/await today. I think it was not prioritized since CDK users cannot use this pattern anyway, since everything has to be done in the constructor, which cannot be async.

We can revisit this for other non-cdk, jsii-based projects in the future. Any work should not require breaking changes.